src/autopas/containers/verletListsCellBased/verletListsCells/VerletListsCells.h

```diff
@@ -115,12 +115,174 @@ class VerletListsCells : public VerletListsLinkedBase<Particle> {
+        1.3);
+
+    // Reset lists. Don't free any memory, only mark as unused.
+    _neighborList.setLinkedCells(&this->_linkedCells);
+    for (auto &cellLists : neighborLists) {
+      for (auto &[particlePtr, neighbors] : cellLists) {
+        particlePtr = nullptr;
+        neighbors.clear();
+      }
+    }
+    neighborLists.resize(cells.size());
+
+    // Helper function to insert a pointer into a list of the base cell.
+    // It considers the cases that neither particle is in the base cell
+    // and in that case finds or creates the appropriate list
+    auto insert = [&](auto &p1, auto indexOfP1sList, auto &p2, auto cellIndex1, auto cellIndexBase,
+                      auto &currentCellsLists) {
+      // Easy case: cell1 is the base cell
+      if (cellIndexBase == cellIndex1) {
+        currentCellsLists[indexOfP1sList].second.push_back(&p2);
+      } else {
+        // Otherwise, check if the base cell already has a list for p1
+        auto iter = std::find_if(currentCellsLists.begin(), currentCellsLists.end(), [&](const auto &pair) {
+          const auto &[particlePtr, list] = pair;
+          return particlePtr == &p1;
+        });
+        // If yes, append p2 to it.
+        if (iter != currentCellsLists.end()) {
+          iter->second.push_back(&p2);
+        } else {
+          // If no, create one, reserve space and emplace p2
+          if (auto insertLoc = std::find_if(currentCellsLists.begin(), currentCellsLists.end(),
+                                            [&](const auto &pair) {
+                                              const auto &[particlePtr, list] = pair;
+                                              return particlePtr == nullptr;
+                                            });
+              insertLoc != currentCellsLists.end()) {
+            auto &[particlePtr, neighbors] = *insertLoc;
+            particlePtr = &p1;
+            neighbors.reserve(listLengthEstimate);
+            neighbors.push_back(&p2);
+          } else {
+            currentCellsLists.emplace_back(&p1, std::vector<Particle *>{});
+            currentCellsLists.back().second.reserve(listLengthEstimate);
+            currentCellsLists.back().second.push_back(&p2);
+          }
+        }
+      }
+    };
+
+    // Vector of offsets from the base cell for the c08 base step
+    // and respective factors for the fraction of particles per cell that need neighbor lists in the base cell.
+    const auto offsetsC08 =
+        VerletListsCellsHelpers::buildC08BaseStep(utils::ArrayUtils::static_cast_copy_array<int>(cellsPerDim));
+
+    // Go over all cells except the very last layer and create lists per base step.
+    // Since there are no loop dependencies merge all for loops and create 10 chunks per thread.
+    AUTOPAS_OPENMP(parallel for collapse(3) schedule(dynamic, std::max(cells.size() / (autopas::autopas_get_max_threads() * 10), 1ul)))
```

Comment:
I experimented with values from 1 to 100 and 10 seemed like a good optimum. However, this might change on a different machine or scenario. Nevertheless, I think we can just leave it like this. If we ever want to investigate tuning of these scheduling parameters, we have to look at all of them anyway.